openmicroscopy/openmicroscopy
# Filter by tag

# What this PR does

Adds an important feature from Insight to webclient: Filter Dataset thumbnails by Tag.
See https://trello.com/c/PNcVcXOt/26-filtering-by-tags-in-webclient

# Testing this PR

1. Tag some images in a Dataset
1. Add Filter -> Tag
1. List of Tags used in the Dataset should be listed in a Select element
1. Choosing Tags adds them to the filter and images without those Tags should be hidden
1. Clicking ```-``` on Tags removes them from the filtering
1. Combine Filtering by Rating and/or Name
1. Add Filter -> Remove all filters  should work as expected

<img width="416" alt="screen shot 2017-11-14 at 22 46 01" src="https://user-images.githubusercontent.com/900055/32809169-c637cb78-c98d-11e7-864b-fd1bf67f503e.png">

TODO: Add robot framework tests.
